packages/product/src/services/product-module-service.ts

```diff
@@ -231,6 +244,119 @@ export default class ProductModuleService<
     return [JSON.parse(JSON.stringify(variants)), count]
   }
 
+  @InjectTransactionManager("baseRepository_")
+  async updateVariants(
+    data: ProductTypes.UpdateProductVariantOnlyDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<ProductTypes.ProductVariantDTO[]> {
+    const variants = await this.listVariants(
+      { id: data.map(({ id }) => id) },
+      { relations: ["options", "options.option"] },
+      sharedContext
+    )
+    const variantsMap = new Map(
+      variants.map((variant) => [variant.id, variant])
+    )
+
+    if (variants.length !== data.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "Cannot update non-existing variants with ids: " +
+          data
+            .filter(({ id }) => !variantsMap.has(id))
+            .map(({ id }) => id)
+            .join(", ")
+      )
+    }
```

Comment:
question: I think riqwan created a diff utils for that? can you check with him please